guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryFormTest.java

```diff
@@ -0,0 +1,121 @@
+package org.guvnor.structure.client.editors.repository.clone;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.guvnor.structure.client.editors.repository.clone.answer.RepositoryServiceAnswer;
+import org.guvnor.structure.client.editors.repository.clone.mock.MockCloneRepositoryPresenter;
+import org.guvnor.structure.organizationalunit.OrganizationalUnit;
+import org.guvnor.structure.organizationalunit.OrganizationalUnitService;
+import org.guvnor.structure.repositories.Repository;
+import org.guvnor.structure.repositories.RepositoryService;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.ErrorCallback;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Matchers.any;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloneRepositoryFormTest {
+
+    @Mock
+    private CloneRepositoryView view;
+    @Mock
+    private Caller<RepositoryService> repoServiceCaller;
+    @Mock
+    private Caller<OrganizationalUnitService> ouServiceCaller;
+    @Mock
+    private OrganizationalUnitService  ouService;
+    @Mock
+    private RepositoryService repoService;
+    @Mock
+    private OrganizationalUnit ouUnit1;
+    @Mock
+    private OrganizationalUnit ouUnit2;
+    @Mock
+    private Repository repository;
+    @Captor
+    private ArgumentCaptor<Boolean> boolArgument;
+    @InjectMocks
+    private CloneRepositoryPresenter presenter = new MockCloneRepositoryPresenter(false);
```

Comment:
Nice, I just learned some thing new. I have been using the constructor for setting up the injected fields. This feels a lot nicer.